all_test.go

```diff
@@ -189,16 +234,45 @@ func isPrefix(prefix, str string) bool {
 	return strings.Index(str, prefix) == 0
 }
 
-//func TestMergeLdap(t *testing.T) {
-//	if (*backend_ldap) {
-//		t.Log("Testing ldap merge")
-//		c := setup1(t)
-//		if err := c.Merge(); err != nil {
-//			t.Errorf("An error occurred while trying to merge ldap users with IMQS users: %v", err)
-//		}
-//		teardown(c)
-//	}
-//}
+func TestMergeLdap(t *testing.T) {
+	if !*backend_ldap {
+		return
+	} else {
+		t.Log("Testing ldap merge")
+		c := setup1(t)
+		defer Teardown(c)
+		if *backend_postgres {
+			userId, err := c.userStore.GetUserIdFromIdentity(joeIdentity)
+			userPermit := setup1_permit()
+			c.permitDB.SetPermit(userId, &userPermit)
+			if err != nil {
+				t.Fatalf("TestMergeLdap failed, could not get userid for: %v", joeIdentity)
+			}
+			setup1_permit()
+			if _, _, err := c.Login(joeIdentity, joePwd); err != nil {
+				t.Fatalf("Login should have successed, but error was : %v", err)
+			}
+		} else {
+			// Add new users to ldap, wait for merge then login
+			ldapTest.AddLdapUser(tomIdentity, tomPwd, "tomh@hotgmail.com", "tom", "hanks", "08467531243")
+			tomhPermit := setup1_permit()
+			c.permitDB.SetPermit(tomUserId, &tomhPermit)
+			//Now we wait for a merge to take place
+			time.Sleep(ldapMergeTickerSeconds*2)
```

Comment:
I don't think we need to test whether the merge ticker actually runs. That code is so predictable that it's almost impossible to get wrong. I mean - the part where we do `for _ := range ticker ...`. And that is actually the thing that's being tested by waiting for the ticker to work. Instead, I would just not run the ticker during tests like this, and instead call the merge function manually. You don't want a unit test like this to take 25 seconds, when the real functionality that's being tested, can be run in under 1 second.
I would keep this test as is, except remove all the waiting. Then, add another test which stresses the possible races, as I described in an earlier comment.